### ISSUE TITLE
Update OpenLayers to latest version

### ISSUE THREAD

**Issue by ZachTRice** [nasa-gibs/worldview#701]
The following features break when updating to the latest version (4.6.x). We need to address each before we can upgrade.
- [x] Vector layers throw errors when zooming out past maximum zoom level. This error seems to have been fixed here: https://github.com/openlayers/openlayers/pull/7125 which was merged into 4.4.0, but 
Error: `Uncaught TypeError: Cannot read property 'getUnits' of undefined`
Note: This error occurs in **4.3.0-4.3.4** but not in 4.4.0+
Fixed in PR #722 
- [x] Granule labels incorrectly positioned #683 
Note: breaks when upgrading to **4.4.x+**
Fixed in PR #722 
- [x] Can not animate custom colormaps
Note: breaking in **4.5.x+**; 
Fixed in PR #699 (Will need to test again with 4.6.x+)
- [x] Zoom indicator (km & mi) are showing the incorrect numbers
Note: This occurs in **4.6.x+**
Pending fix in https://github.com/openlayers/openlayers/pull/7700 (waiting for a release with this change)

**Comment by localjo** [comment]
Updating also causes a bug with the map scale:

![screen shot 2018-01-18 at 12 58 31 pm](https://user-images.githubusercontent.com/516388/35114408-97155aca-fc52-11e7-9a6a-a43b2200d439.png)

**Comment by PacoDu** [comment]
I was wondering: What made you choose OpenLayer over Leaflet ? Just for curiosity.

**Comment by rboller** [comment]
Hi @PacoDu, we made that decision awhile ago after some careful consideration - Leaflet would've likely also worked for us, though OpenLayers was a better fit in general for our needs.  From what I recall, one of the main points of consideration was better map projection support in OL since we needed to support polar stereographic projections along with geographic.  There also appeared to be a trend of [increasing contributions to OL](https://github.com/openlayers/openlayers/graphs/contributors) and [decreasing contributions to Leaflet](https://github.com/Leaflet/Leaflet/graphs/contributors) during that time.

That said, Leaflet is of course still a great library and we have set of demos with a legacy version of it (v0.7.3) with GIBS here:
https://github.com/nasa-gibs/gibs-web-examples#live-examples

If you're interested in Leaflet and GIBS, you're certainly welcome to bring those examples up to date!

**Comment by ZachTRice** [comment]
OpenLayers needs to be upgraded before we can support vector layers. Version 4.3.4 has a bug where non-native resolutions (i.e. zoomed out further than specified in GC). This was fixed in later versions but later versions cause conflicts with other parts of the application. See: https://github.com/openlayers/openlayers/issues/6946

**Comment by ZachTRice** [comment]
Note: 4.6.x will likely be the last 4.x version (as indicated in the 4.6.0 release notes). 

Here is a run down of what is coming in 5.x and how we can prepare for it: https://github.com/openlayers/openlayers/wiki/OpenLayers-v5.x

Looks like we are in pretty good shape but we will need to review some of our module imports to match their new naming convention.

**Comment by ZachTRice** [comment]
These 3 lines of code that were added in 4.6.0 are what is causing our issue with the map scale in 4.6.x+:
https://github.com/openlayers/openlayers/pull/7462/files#diff-2924cafbc70baae54a9effb87d39a1f8

Removing these lines & rebuilding the app fixes the issue. Now I just need to figure out if we can work around this or if I need to put in a fix to OpenLayers.

**Comment by ZachTRice** [comment]
Ah, seems like someone else discovered this issues as well. A fix was put in about two weeks ago:
https://github.com/openlayers/openlayers/issues/7694

This PR https://github.com/openlayers/openlayers/pull/7700 fixes the issue (I converted the code changes in that PR to `ol` syntax and rebuilt the app; tested & works). That PR has been merged into their master branch but has not yet been put into a release.

So, I am going to label this issue now as `external dependency` as we can revisit once https://github.com/openlayers/openlayers/pull/7700 is put into a release.

**Comment by localjo** [comment]
Openlayers v5.x is looking good! Nice that they provide us with an upgrade guide. :) I agree, it looks like the change should be easy for us to make. They haven't scheduled a date for their release, but it looks like we can track their progress on the milestone: https://github.com/openlayers/openlayers/milestone/36

**Comment by ZachTRice** [comment]
This work is not completely done. Re-opening this issue until every subtask is completed.

**Comment by minniewong** [comment]
@ZachTRice  - Review OpenLayers 4.6.5

**Comment by ZachTRice** [comment]
4.6.5 does not fix the issue with the zoom indicator. We will need to wait until 5.0 before upgrading.

**Comment by ZachTRice** [comment]
We will need to wait on upgrading to Openlayers 5.x until we have a solution in place to handle ES Module packages. Since Openlayers 5.x is written with ES6 import/exports and not commonjs, it needs to be compiled (babelified) first. Our current browserify/babel setup does not babelify the node_modules, only the js directory. We can add node_modules to the babel path but this will take significantly more time to compile and re-compile the application. Ideally, tree-shaking should take place on node_modules first to only pull in the imports from the node_modules directory that are needed. This is handled well by other task runners like webpack, rollup, or parcel.

With plans to implement webpack in the future, it would make sense to upgrade to OL 5.x after that has occurred.
